modules/omsBidAdapter.js

```diff
@@ -0,0 +1,321 @@
+import {
+  isArray,
+  getWindowTop,
+  getUniqueIdentifierStr,
+  deepSetValue,
+  logError,
+  logWarn,
+  createTrackPixelHtml,
+  getWindowSelf,
+  isFn,
+  isPlainObject, getBidIdParameter,
+} from '../src/utils.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+import {config} from '../src/config.js';
+import {ajax} from '../src/ajax.js';
+
+const BIDDER_CODE = 'oms';
+const URL = 'https://rt.marphezis.com/hb';
+const TRACK_EVENT_URL = 'https://rt.marphezis.com/prebid'
+
+export const spec = {
+  code: BIDDER_CODE,
+  gvlid: 883,
+  supportedMediaTypes: [BANNER],
+  isBidRequestValid,
+  buildRequests,
+  interpretResponse,
+  onBidderError,
+  onTimeout,
+  onBidWon,
+  getUserSyncs,
+};
+
+function buildRequests(bidReqs, bidderRequest) {
+  try {
+    const impressions = bidReqs.map(bid => {
+      let bidSizes = bid?.mediaTypes?.banner?.sizes || bid.sizes;
+      bidSizes = ((isArray(bidSizes) && isArray(bidSizes[0])) ? bidSizes : [bidSizes]);
+      bidSizes = bidSizes.filter(size => isArray(size));
+      const processedSizes = bidSizes.map(size => ({w: parseInt(size[0], 10), h: parseInt(size[1], 10)}));
+
+      const element = document.getElementById(bid.adUnitCode);
+      const minSize = _getMinSize(processedSizes);
+      const viewabilityAmount = _isViewabilityMeasurable(element) ? _getViewability(element, getWindowTop(), minSize) : 'na';
+      const viewabilityAmountRounded = isNaN(viewabilityAmount) ? viewabilityAmount : Math.round(viewabilityAmount);
+
+      const imp = {
+        id: bid.bidId,
+        banner: {
+          format: processedSizes,
+          ext: {
+            viewability: viewabilityAmountRounded
+          }
+        },
+        tagid: String(bid.adUnitCode)
+      };
+
+      const bidFloor = _getBidFloor(bid);
+
+      if (bidFloor) {
+        imp.bidfloor = bidFloor;
+      }
+
+      return imp;
+    })
+
+    const referrer = bidderRequest?.refererInfo?.page || '';
+    const publisherId = getBidIdParameter('publisherId', bidReqs[0].params);
+
+    const payload = {
+      id: getUniqueIdentifierStr(),
+      imp: impressions,
+      site: {
+        domain: bidderRequest?.refererInfo?.domain || '',
+        page: referrer,
+        publisher: {
+          id: publisherId
+        }
+      },
+      device: {
+        devicetype: _getDeviceType(),
+        w: screen.width,
+        h: screen.height
+      },
+      tmax: bidderRequest?.timeout
+    };
+
+    if (bidderRequest?.gdprConsent) {
+      deepSetValue(payload, 'regs.ext.gdpr', +bidderRequest.gdprConsent.gdprApplies);
+      deepSetValue(payload, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    }
+
+    if (bidderRequest?.uspConsent) {
+      deepSetValue(payload, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    }
+
+    if (config.getConfig('coppa') === true) {
+      deepSetValue(payload, 'regs.coppa', 1);
```

Comment:
Please get this from the request

Reply:
Fixed, now we getting it from response

Reply:
you seem to be falling back to the request, please don't call getConfig

Reply:
Fixed